### PR TITLE
Update CODEOWNERS for IoT Hub Service SDK owners

### PR DESCRIPTION
**Pull request by andyk-ms**
Update the owners for IoT Hub service team.

# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by konrad-jamrozik** [comment]
@andyk-ms thank you for this PR! How about we close it and I incorporate the changes in my PR, to avoid PR conflicts?

- https://github.com/Azure/azure-sdk-for-net/pull/33826

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -392,7 +392,7 @@
 #/<NotInRepo>/            @madhurinms
```
[the remark this answers is not in the document]
Comment (konrad-jamrozik) [reply .github/CODEOWNERS:393]: @jsquire @andyk-ms I can take care of this. See also [my comment](https://github.com/Azure/azure-sdk-for-net/pull/33854#issuecomment-1414433168) about closing this PR and incorporating it into my PR.

**Comment by andyk-ms** [comment]
Sure I will close this, and please include the changes on your change.

**Comment by konrad-jamrozik** [comment]
Change applied here:
- https://github.com/Azure/azure-sdk-for-net/pull/34103/files

**Pull request closed without merging**